extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/SecurityCheck.java

```diff
@@ -5,5 +5,8 @@
 import io.quarkus.security.identity.SecurityIdentity;
 
 public interface SecurityCheck {
+
     void apply(SecurityIdentity identity, Method method, Object[] parameters);
+
+    void apply(SecurityIdentity identity, MethodDescription methodDescription, Object[] parameters);
```

Comment:
I preferred to not a default in the existing method that would delegate to this because that would result in unnecessary allocations of `MethodDescription` when in most cases it is not needed at all.